python/eos/plot/plotter.py

```diff
@@ -671,6 +672,181 @@ def plot(self):
+                    yobservable = (self.yobservable, self.ykinematics, self.yoptions)
+
+                    if xobservable not in observables:
+                        raise ValueError('x-axis observable {} not contained in constraint {}'.format(self.xobservable, self.constraint))
+
+                    if yobservable not in observables:
+                        raise ValueError('y-axis observable {} not contained in constraint {}'.format(self.yobservable, self.constraint))
+
+                    xidx = observables.index(xobservable)
+                    yidx = observables.index(yobservable)
+
+                    means  = np.array(constraint['means'])
+                    xmean  = means[xidx]
+                    ymean  = means[yidx]
+
+                    cov    = np.array(constraint['covariance'])
+                    xsigma = np.sqrt(covariance[xidx, xidx])
+                    ysigma = np.sqrt(covariance[yidx, yidx])
+                    rho    = cov[xidx, yidx] / (xsigma * ysigma)
+                else:
+                    observables = list(zip(constraint['observables'], constraint['kinematics'], constraint['options']))
+
+                    xobservable = (self.xobservable, self.xkinematics, self.xoptions)
+                    yobservable = (self.yobservable, self.ykinematics, self.yoptions)
+
+                    if xobservable not in observables:
+                        raise ValueError('x-axis observable {} not contained in constraint {}'.format(self.xobservable, self.constraint))
+
+                    if yobservable not in observables:
+                        raise ValueError('y-axis observable {} not contained in constraint {}'.format(self.yobservable, self.constraint))
+
+                    xidx = observables.index(xobservable)
+                    yidx = observables.index(yobservable)
+
+                    means  = np.array(constraint['means'])
+                    xmean  = means[xidx]
+                    ymean  = means[yidx]
+
+                    sigmas = np.sqrt(
+                          (np.array(constraint['sigma-stat-hi'])**2 + np.array(constraint['sigma-stat-lo'])**2) / 4.0
+                        +  np.array(constraint['sigma-sys'])**2
+                    )
+                    xsigma = sigmas[xidx]
+                    ysigma = sigmas[yidx]
+
+                    rho    = np.array(constraint['correlations'])[xidx, yidx]
+
+                sigmas = [1.0, 2.0, 3.0]
+                alphas = np.linspace(0.0, self.alpha, len(sigmas) + 1)[1:]
+                for sigma, alpha in zip(sorted(sigmas, reverse=True), alphas):
+                    xwidth = 2.0 * xsigma * np.sqrt(1.0 + rho) * sigma
+                    ywidth = 2.0 * ysigma * np.sqrt(1.0 - rho) * sigma
+
+                    ellipse = matplotlib.patches.Ellipse((xmean, ymean), width=xwidth, height=ywidth, angle=45,
+                                          alpha=alpha, color=self.color, linewidth=1, fill=True)
```

Comment:
I don't understand your ellipse, why is the angle set at 45? Shouldn't this angle depend on the correlation?

Reply:
I will investigate. Might have implemented the matplotlib example incorrectly.

Reply:
I made a mistake. The various transformations do not commute. Fixed.